app/models/manageiq/providers/autosde/storage_manager/event_catcher/stream.rb

```diff
@@ -17,6 +17,12 @@ def stop
   end
 
   def poll_events(&block)
+    physical_storages = []
+    while physical_storages.empty?
+      physical_storages = PhysicalStorage.where(:ems_id => @ems.id)
+      sleep(poll_sleep)
+    end
```

Comment:
Hey @Oferlis so this has a number of issues with it that we can get into but just so I understand, is the problem you're trying to solve that events are caught by the event catcher and saved by the event handler before the refresh_worker finishes the first refresh leaving events without an associated `physical_storage_id`?  Usually the way this is solved is by adding a `post_create` action to the model which reconnects events which were saved before the associated record is created.  See `VmOrTemplate#reconnect_events` for an example of this https://github.com/ManageIQ/manageiq/blob/master/app/models/vm_or_template.rb#L862

This does mean that there has to be something unique saved to the event_streams record that can be used, so I think we'd want to save the ems_ref then we can use that for reconnecting in the future.

Reply:
Hey @agrare when should I call the post_create method? 
Should it be a part of the event polling loop?

Reply:
@Oferlis it is part of refresh (because at that point you know the inventory records are created).

Reply:
https://github.com/ManageIQ/manageiq-providers-amazon/blob/master/app/models/manageiq/providers/amazon/cloud_manager/refresher.rb#L15-L18 is an example of telling refresh to post-process those classes
Then you have to define a `self.post_refresh_ems` method on the class like https://github.com/ManageIQ/manageiq/blob/master/app/models/vm_or_template.rb#L1068